crates/flux-fhir-analysis/src/lib.rs

```diff
@@ -194,11 +195,32 @@ fn assoc_refinements_of(
     Ok(rty::AssocRefinements { items: predicates })
 }
 
-fn assoc_refinement_def(
+fn default_assoc_refinement_def(
+    genv: GlobalEnv,
+    trait_id: LocalDefId,
+    name: Symbol,
+) -> QueryResult<Option<rty::EarlyBinder<rty::Lambda>>> {
+    let impl_id = trait_id.to_def_id();
```

Comment:
The idea is not to use `trait_id` at all before calling `maybe_extern_id`. If you _really_ need the local id converted to a `DefId` you need to do `trait_id.local_id().to_def_id()`, that being said, I think the `impl_id` in `InvalidAssocReft` should be the resolve id.

Reply:
We should also change the name of `impl_id` in the definition of `QueryErr::InvalidAssocReft` to be `container_id` or `container_def_id` now that it can refer to a trait.